packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,423 @@
+    function removeAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _removeAddresses(_members);
+
+        // Check if the new addresslist has become shorter than the current minimum number of approvals required.
+        uint256 newAddresslistLength = addresslistLength();
+        if (newAddresslistLength < minApprovals_) {
+            revert MinApprovalsOutOfBounds({limit: newAddresslistLength, actual: minApprovals_});
+        }
+    }
+
+    /// @notice Creates a new majority voting proposal.
+    /// @param _metadata The metadata of the proposal.
+    /// @param _actions The actions that will be executed after the proposal passes.
+    /// @param _approveProposal If `true`, the sender will approve the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the vote cast. The call does not revert if early execution is not possible.
+    /// @return proposalId The ID of the proposal.
+    function createProposal(
+        bytes calldata _metadata,
+        IDAO.Action[] calldata _actions,
+        bool _approveProposal,
+        bool _tryExecution
+    ) external returns (uint256 proposalId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1;
+
+        if (!isListedAtBlock(_msgSender(), snapshotBlock)) {
+            revert ProposalCreationForbidden(_msgSender());
+        }
+
+        proposalId = proposalCounter.current();
+
+        // Create the proposal
+        Proposal storage proposal_ = proposals[proposalId];
+
+        proposal_.open = true;
+        proposal_.parameters.snapshotBlock = snapshotBlock;
+        proposal_.parameters.minApprovals = minApprovals_;
+        proposal_.tally.addresslistLength = addresslistLengthAtBlock(snapshotBlock); // TODO https://aragonassociation.atlassian.net/browse/APP-1417
+
+        for (uint256 i; i < _actions.length; ) {
+            proposal_.actions.push(_actions[i]);
+            unchecked {
+                ++i;
+            }
+        }
+
+        _incrementProposalCount();
+
+        if (_approveProposal) {
+            approve(proposalId, _tryExecution);
+        }
+
+        emit ProposalCreated({
+            proposalId: proposalId,
+            creator: _msgSender(),
+            metadata: _metadata,
+            actions: _actions
+        });
+    }
+
+    /// @notice Approves and, optionally, executes the proposal.
+    /// @param _proposalId The ID of the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the approval cast. The call does not revert if execution is not possible.
+    function approve(uint256 _proposalId, bool _tryExecution) public {
+        address approver = _msgSender();
+        if (!_canApprove(_proposalId, approver)) {
+            revert ApprovalCastForbidden(_proposalId, approver);
+        }
+
+        Proposal storage proposal_ = proposals[_proposalId];
+
+        proposal_.tally.approvals += 1;
```

Comment:
could use `unchecked` because `approvals` is a `uint256` and is nowhere increased by more than 1. Additionally overflowing this uint would imply that that list is longer than the max of `uint256`

Reply:
Done in 14addab33026d06f295f963a2f55bad7b170f91a

should we also do it in `AddresslistVoting` (in a future PR)? https://github.com/aragon/core/blob/d70ed76cc828202d9c9f4dde0690a9fd12619604/packages/contracts/contracts/voting/addresslist/AddresslistVoting.sol#L156-L172
Should it be done for the addition **and** subtraction of the pre-existing vote (that can be assumed to be >=1)?

Reply:
for additions I would but not for subtractions because making sure it is >= 1 is usally more gas intensive

Reply:
Since we now because of `state == VoteOption.Yes` that the voter has previously voted `yes`, this implies that the counter is at least `>=1` so we don't need do this check explicitly.
However, this could be a sacrifice in security in case an underflow could happen.

Reply:
In a call, @novaknole @mathewmeconry and I agreed to do all the gas optimizations globally in a future PR before the product is launched.